contracts/ClearingHouse.sol

```diff
@@ -511,18 +511,30 @@ contract ClearingHouse is
     function quitMarket(address trader, address baseToken) external override returns (uint256 base, uint256 quote) {
         // CH_MNC: Market not closed
         require(IBaseToken(baseToken).isClosed(), "CH_MNC");
-        // CH_HOICM: Has order in closed market
-        require(_getOpenOrderIds(trader, baseToken).length == 0, "CH_HOICM");
-
-        int256 positionSize = _getTakerPositionSafe(trader, baseToken);
 
         _settleFunding(trader, baseToken);
 
+        bytes32[] memory orderIds = _getOpenOrderIds(trader, baseToken);
+
+        // remove all orders in internal function
+        if (orderIds.length > 0) {
+            _removeAllLiquidity(trader, baseToken, orderIds);
+        }
+
+        int256 positionSize = _getTakerPosition(trader, baseToken);
+
+        // if position is 0, no need to do settlement accounting
+        if (positionSize == 0) {
+            return (0, 0);
+        }
+
         (int256 positionNotional, int256 openNotional, int256 realizedPnl, uint256 closedPrice) =
             IAccountBalance(_accountBalance).settlePositionInClosedMarket(trader, baseToken);
 
         emit PositionClosed(trader, baseToken, positionSize, positionNotional, openNotional, realizedPnl, closedPrice);
 
+        _settleBadDebt(trader);
```

Comment:
There's a line in `Valut.settleBadDebt()`: `require(_collateralTokensMap[trader].length != 0)`, which requires users to not have non-USDC collateral.
Therefore, won't this `quitMarket()` get stuck if a user still has non-USDC collateral?

Reply:
yes, so if the user still has bad debt after quitMaket (`_settleBadDebt()` didn't work when quitMarket()), we will need to liquidate his collateral via `Vault.liquidateCollateral()` and will trigger `settleBadDebt()` again

Reply:
So do we put this logic in BE? and this is for... saving contract bytecode size?
If this is the case, suggest writing a test case to demonstrate that this scenario will fail and add a comment to let others know that we have to first `liquidateCollateral` and then call `quitMarket`!

Reply:
We don't need to add new logic on BE (we already have liquidated collateral bot), and we don't need to call `liquidateCollateral` first because `_settleBadDebt` will only care about user's `accountValue`. we will call `_settleBadDebt` in `CH.liquidatePosition`, `CH.quitMarket`, and `Vault.liquidateCollateral`. If that call satisfied the condition of settling bad debt, we would settle it. These three actions didn't have any order dependencies (I think we can discuss in gather/google meet, it will be more clearly to explain)

Reply:
Got it!